Src/Test.UnitTests.Sarif.PatternMatcher.Cli/AnalyzeCommandTests.cs

```diff
@@ -277,6 +280,98 @@ public void AnalyzeCommand_ShouldThrowExceptionForConflictingFileSizeOptions()
             Program.InstantiatedAnalyzeCommand.Should().NotBeNull();
         }
 
+
+        [Fact]
+        public void AnalyzeCommand_FailureLevelShouldBeDefaultUnlessDynamicallyValidated()
+        {
+            var sb = new StringBuilder();
+
+            var testCases = new[] {
+                new {
+                    jsonLevel = FailureLevel.Note,
+                    dynamicValidationEnabled = false,
+                    expectedNoValidation = FailureLevel.Note,
+                    expectedStaticOnly = FailureLevel.Note,
+                    expectedStaticDynamic = FailureLevel.Warning
+                },
+                new {
+                    jsonLevel = FailureLevel.Warning,
+                    dynamicValidationEnabled = false,
+                    expectedNoValidation = FailureLevel.Warning,
+                    expectedStaticOnly = FailureLevel.Warning,
+                    expectedStaticDynamic = FailureLevel.Warning
+                },
+                new {
+                    jsonLevel = FailureLevel.Error,
+                    dynamicValidationEnabled = false,
+                    expectedNoValidation = FailureLevel.Error,
+                    expectedStaticOnly = FailureLevel.Error,
+                    expectedStaticDynamic = FailureLevel.Warning
+                },
+                new {
+                    jsonLevel = FailureLevel.Note,
+                    dynamicValidationEnabled = true,
+                    expectedNoValidation = FailureLevel.Note,
+                    expectedStaticOnly = FailureLevel.Note,
+                    expectedStaticDynamic = FailureLevel.Error
+                },
+                new {
+                    jsonLevel = FailureLevel.Warning,
+                    dynamicValidationEnabled = true,
+                    expectedNoValidation = FailureLevel.Warning,
+                    expectedStaticOnly = FailureLevel.Warning,
+                    expectedStaticDynamic = FailureLevel.Error
+                },
+                new {
+                    jsonLevel = FailureLevel.Error,
+                    dynamicValidationEnabled = true,
+                    expectedNoValidation = FailureLevel.Error,
+                    expectedStaticOnly = FailureLevel.Error,
+                    expectedStaticDynamic = FailureLevel.Error
+                }
+            };
+
+            foreach (var testCase in testCases)
+            {
+                string definitionsText = GetSingleLineRuleDefinitionFailureLevel(testCase.jsonLevel);
+                string testScenarioName = "NoValidation";
+                SarifLog logFile = RunAnalyzeCommandWithDynamicValidation(definitionsText,
+                                                                          testScenarioName,
+                                                                          testCase.dynamicValidationEnabled);
+
+                ValidateFailureLevelByValidationType(expectedFailureLevel: testCase.expectedNoValidation,
```

Comment:
>expectedFailureLevel [](http://example.com/codeflow?start=53&length=20)

please consider my suggested renames. if you took the suggestion, i think you'd agree that the test case property name itself is sufficient documentation of its purpose and you wouldn't need a named parameter. that would be the goal.